### PR TITLE
[MM-45625] Do not enable the Keyboard Search button unless there is text in the search bar

### PR DESCRIPTION
**Pull request by jfrerich**
#### Summary
This PR disables the keyboard search button until there is text in the search bar text input.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45625

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
This PR was tested on: 

#### Release Note

```release-note

```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

**Review by enahum — CHANGES_REQUESTED** [review]
This applies to all the search bars in the app instead of only applying in the search screen as in the ticket

**Comment by enahum** [comment]
Unless we want to apply this to every search bar changes should be made. If we **do** want this to apply to all search bars, then I'll approve it

**Review by larkox — APPROVED** [review]
Apart of enahum's comments, LGTM

File: app/screens/home/search/search.tsx [hunk 1]
```diff
@@ -320,90 +320,91 @@ const SearchScreen = ({teamId, teams}: Props) => {
     return (
         <FreezeScreen freeze={!isFocused}>
             <NavigationHeader
-                isLargeTitle={true}
-                showBackButton={false}
-                title={intl.formatMessage({id: 'screen.search.title', defaultMessage: 'Search'})}
+                blurOnSubmit={true}
+                defaultValue={searchValue}
+                enablesReturnKeyAutomatically={true}
```
Comment (jfrerich) [inline app/screens/home/search/search.tsx:325]: I've added the new prop at this level now.

**Comment by jfrerich** [comment]
> Unless we want to apply this to every search bar changes should be made. If we **do** want this to apply to all search bars, then I'll approve it
@enahum, I moved the prop to the Search Screen level just to be safe. 

I also sorted all the component props alphabetically because I didn't do that way back when I did the search screen.

I tested how `Find Channels` handles the empty search and it closes the keyboard if you search and a search term does not exist.  I propose to create a ticket if we want to have UX review all the behaviors or empty search terms across the product.

**Comment by enahum** [comment]
> > Unless we want to apply this to every search bar changes should be made. If we **do** want this to apply to all search bars, then I'll approve it
> > @enahum, I moved the prop to the Search Screen level just to be safe.
> 
> I also sorted all the component props alphabetically because I didn't do that way back when I did the search screen.
> 
> I tested how `Find Channels` handles the empty search and it closes the keyboard if you search and a search term does not exist. I propose to create a ticket if we want to have UX review all the behaviors or empty search terms across the product.

Then I cannot approve this PR

**Comment by matthewbirtch** [comment]
> Unless we want to apply this to every search bar changes should be made. If we **do** want this to apply to all search bars, then I'll approve it

I would think that this _should_ apply to every instance of the search bar. I can't think of a use case for enabling the search button when the field is empty.

**Comment by jgilliam17** [comment]
Thanks @jfrerich 
iPhone shows disabled Enter button when search bar is empty, but on iPad I can still tap Enter.
On Android, Enter also seems enabled.
IMG_5D9DFF655029-1
Screen Shot 2023-01-24 at 4 42 12 PM
IMG_6C7B80A35906-1

**Comment by enahum** [comment]
closing for now

**Pull request closed without merging**